src/cljs/s3_beam/client.cljs

```diff
@@ -96,35 +96,55 @@
 (defn upload-file
   "Take the signature information `upload-info`, and a response channel `ch`,
   and does the S3 upload.
-  Responds on the response channel with a map with:
+  Upon success it responds on the response channel with a map with:
+    :type - :success for success events
     :file - the File that was uploaded
     :response - a map of S3 metadata: :key, :location, :bucket, :etag
     :identifier - an identifier pass-through value for caller to identify the file with
   If the upload-info has an :error-code key because the signing failed, skips the xhr/send of the file,
-  and simply replies with the error messages and codes from the failed upload-info sign attempt."
-  [{:keys [f identifier form-data upload-url] :as upload-info} ch]
+  and simply replies with the error messages and codes from the failed upload-info sign attempt.
+  If progress-events are enabled, it responds on the response channel with a map with
+    :type - :progress
+    :file - the File that was uploaded
+    :bytes-sent - the size of the already uploaded potion in bytes
+    :bytes-total - the total file size in bytes
+    :identifier - an identifier pass-through value for caller to identify the file with"
+  [progress-events {:keys [f identifier form-data upload-url] :as upload-info} ch]
```

Comment:
I think instead of passing `progress-events` as argument here we should pass a map so we can extend the number of options passed without introducing backwards incompatible (fn-arity) changes.
```clojure
[{:keys [progress-events] :as opts} ... ]
```